Update cmdlet to support arc gateway

xw-zhang24 opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR. If you have an ongoing or finished cmdlet design, please paste the link below. -->

The PR aims to add support to enable arc gateway feature on ConnectedKubernetes cmdlet during resource creation and update. Due to limit of current swagger design on `PATCH`, we introduced `Set-AzConnectedKubernetes` to allow modification on arc gateway feature on existing resources following the best practice. Design and discussion are available in [this doc](https://microsoft.sharepoint.com/:w:/t/NSODevTeam/EcCDgRYkophJn8QcWVDeaLABYxbMMyzer1KcOkayN-FKkQ?e=WP09cM)

We suggest to put following bullets in changelog:
- Added support for arc gateway feature in cmdlet New-AzConnectedKubernetes 
- Added Set-AzConnectedKubernetes cmdlet to support updateing arc gateway features on existing resource 

We are waiting for dependency services to be released first in late September, so the targeting release for now will be October.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

xw-zhang24 commented:
@msJinLei  and other reviews, 

The pipeline currently fails at build due to autorest failure on `Storage`, which is totally unrelated to the PR change. In fact, the failing module, along with other module, only started to be included at build from [commit](https://github.com/Azure/azure-powershell/pull/26026/commits/37956893a05c0b4801be367fc20e91cb23a0b54e) which adds analysis exception, I wonder if that triggers all modules to build.  Regarding the failure itself, I'm not familiar with the failing module itself,  but I've tried to build the storage module locally, it failed with autorest 4.0 but passed with autorest 3.0.  Is there anyway to build without unrelated module, or configure some modules to build on different autorest version? If these are not options, how to make the pipeline happy again? Thanks,

msJinLei commented:
Lei is following up the build issue

msJinLei commented on src/ConnectedKubernetes/ConnectedKubernetes.Autorest/docs/Set-AzConnectedKubernetes.md:
````diff
@@ -0,0 +1,635 @@
+
+### SetExpanded (Default)
+```
+Set-AzConnectedKubernetes -ClusterName <String> -ResourceGroupName <String> -Location <String>
````
Please use singular name

xw-zhang24 replied:
@msJinLei  We kept the plural form in the name to maintain consistency with the other commands in this module.:
```
Get-AzConnectedKubernetes
New-AzConnectedKubernetes
Remove-AzConnectedKubernetes
Update-AzConnectedKubernetes
```

We've also put an exception to suppress the analysis check for plural in [this commit](https://github.com/Azure/azure-powershell/pull/26026/commits/37956893a05c0b4801be367fc20e91cb23a0b54e)

msJinLei reviewed (changes requested):
Please provide the change log for your PR

xw-zhang24 commented:
> Please provide the change log for your PR

@msJinLei Thanks for your review and comments.

As the change is autorest based, we have put changelog in PR description as required, but happy to duplicate below.

We suggest to put following bullets in changelog:

```
Version 0.11.0
- Added support for arc gateway feature in cmdlet New-AzConnectedKubernetes
- Added Set-AzConnectedKubernetes cmdlet to support updateing arc gateway features on existing resource
```
